pkg/resources/network_policy.go

```diff
@@ -0,0 +1,319 @@
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the network policy.",
+	},
+	"set_for_account": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Specifies whether this network policy should be applied globally to your Snowflake account<br><br>**Note:** The Snowflake user running `terraform apply` must be on an IP address allowed by the network policy to set that policy globally on the Snowflake account.<br><br>Additionally, a Snowflake account can only have one network policy set globally at any given time. This resource does not enforce one-policy-per-account, it is the user's responsibility to enforce this. If multiple network policy resources have `set_for_account: true`, the final policy set on the account will be non-deterministic.",
+		Default:     false,
+		ForceNew:    true,
+	},
+	"users": {
+		Type:        schema.TypeSet,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Optional:    true,
+		Description: "Specifies which users this network policy should be applied to.<br><br>**Note**: The network policy resource creates exclusive assignments of policies to users. Any users that have a network policy assigned outside of that policy's Terraform resource definition will have the policy unset.<br><br>For safety, you should *only* ever assign/unset network policies to users via Terraform.<br><br>Additionally, a Snowflake user can only have one network policy applied to it. This resource does not enforce one-policy-per-user, it is the user's responsibility to enforce this. If a user appears in multiple network policy resource definitions, the final policy set on the user will be non-deterministic.",
+		ForceNew:    true,
+	},
+}
+
+// NetworkPolicy returns a pointer to the resource representing a network policy
+func NetworkPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateNetworkPolicy,
+		Read:   ReadNetworkPolicy,
+		Update: UpdateNetworkPolicy,
+		Delete: DeleteNetworkPolicy,
+
+		Schema: networkPolicySchema,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+// CreateNetworkPolicy implements schema.CreateFunc
+func CreateNetworkPolicy(data *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+
+	name := data.Get("name").(string)
+	allowedIps := snowflake.IpListToString(expandStringList(data.Get("allowed_ip_list").(*schema.Set).List()))
+	blockedIps := snowflake.IpListToString(expandStringList(data.Get("blocked_ip_list").(*schema.Set).List()))
+	comment := data.Get("comment").(string)
+
+	err := ensureUserAlterPrivileges(data, meta)
+	if err != nil {
+		return err
+	}
+
+	stmt := fmt.Sprintf(`CREATE NETWORK POLICY "%v" ALLOWED_IP_LIST=%v BLOCKED_IP_LIST=%v COMMENT="%v"`, name, allowedIps, blockedIps, comment)
```

Comment:
Can we move this sql generation to the snowflake package with the rest of the SQL?